[New Model]: support GTE NewModel (#17986)
vllm-project/vllm: docs/source/models/supported_models.md

  * ✅︎
  * ✅︎
- * `GteModel`

>   * Snowflake Arctic-Embed-2.0-M

Reviewer: ```suggestion
  * Arctic Embed 2.0
```

This is what they used in their model card

Author: Arctic Embed 2.0 has two model types: 2.0-M for GteModel, 2.0-L for XLMRobertaModel

Reviewer: I see... this is quite confusing

Reviewer: Maybe name it Arctic Embed 2.0 M?

>   * Snowflake names GTE `NewModel` as `GteModel`,  [paper](https://arxiv.org/abs/2412.04506), [hf](https://huggingface.co/Snowflake/snowflake-arctic-embed-m-v2.0)

Author: We use two open-source pretrained encoder models: gte-multilingual-mlm-base (Zhang et al.,
2024) for medium size. gte-multilingual-mlm-base longer

>   * Snowflake names GTE `NewModel` as `GteModel`,  [paper](https://arxiv.org/abs/2412.04506), [hf](https://huggingface.co/Snowflake/snowflake-arctic-embed-m-v2.0)

Author: call GTE `NewModel` as GTE Multilingual Base in table 1

>   * Snowflake names GTE `NewModel` as `GteModel`,  [paper](https://arxiv.org/abs/2412.04506), [hf](https://huggingface.co/Snowflake/snowflake-arctic-embed-m-v2.0)

Author: It seems unnecessary to write about its relationship with Gte model.

how about 
  * Snowflake Arctic-Embed-2.0-M [paper](https://arxiv.org/abs/2412.04506), [hf](https://huggingface.co/Snowflake/snowflake-arctic-embed-m-v2.0)

>   * Snowflake names GTE `NewModel` as `GteModel`,  [paper](https://arxiv.org/abs/2412.04506), [hf](https://huggingface.co/Snowflake/snowflake-arctic-embed-m-v2.0)

Author: how about now

  * `Snowflake/snowflake-arctic-embed-m-v2.0`.
  *
  * ︎
- * `GteNewModel`
  * mGTE-TRM (see note)
  * `Alibaba-NLP/gte-multilingual-base`, etc.
  * ︎
  * ︎
- * `ModernBertModel`

>   * ModernBertModel-based

Reviewer: ```suggestion
  * ModernBERT-based
```

  * `Alibaba-NLP/gte-modernbert-base`, etc.
  * ︎
  * ︎
- * `NomicBertModel`

>   * NomicBertModel-based

Reviewer: ```suggestion
  * Nomic BERT
```

  * `Snowflake/snowflake-arctic-embed-m-v2.0`.
  *
  * ︎
- * `ModernBertModel`
  * ModernBertModel
  * `Alibaba-NLP/gte-modernbert-base`, etc.
  * ︎
  * ︎
- * `GteNewModel`

>   * GTE NewModel

Reviewer: Can you update the names to use natural language?

Reviewer: The raw class name is in the previous column already

Author: how about now

`jinaai/jina-embeddings-v3` supports multiple tasks through lora, while vllm temporarily only supports text-matching tasks by merging lora weights.
:::

:::{note}

> The name `NewModel` is too generic, you should set `--hf-overrides '{"architectures": ["GteNewModel"]}'` specify the use of the GteNewModel architecture.

Reviewer: ```suggestion
The name `NewModel` is too generic, you should set `--hf-overrides '{"architectures": ["GteNewModel"]}'` specify the use of the `GteNewModel` architecture.
```

> The 2-gen GTE model(mGTE-TRM) is named `NewModel`. The name `NewModel` is too generic, you should set `--hf-overrides '{"architectures": ["GteNewModel"]}'` specify the use of the `GteNewModel` architecture.

Reviewer: ```suggestion
The second-generation GTE model (mGTE-TRM) is named `NewModel`. The name `NewModel` is too generic, you should set `--hf-overrides '{"architectures": ["GteNewModel"]}'` specify the use of the `GteNewModel` architecture.
```
